homeassistant/components/obihai/entity.py

```diff
@@ -1,18 +1,31 @@
 """Obihai Entity base class."""
 from pyobihai import PyObihai
 
-from .const import OBIHAI
+from homeassistant.helpers.entity import DeviceInfo
+
+from .const import DOMAIN, OBIHAI
 
 
 class ObihaiEntity:
     """Obihai Entity base Class."""
 
+    _attr_has_entity_name = True
+
     def __init__(self, pyobihai: PyObihai, serial: str, service_name: str) -> None:
         """Initialize monitor sensor."""
         self._pyobihai = pyobihai
         self._service_name = service_name
+        self._serial = serial
         self._unique_id = f"{serial}-{self._service_name}"
-        self._name = f"{OBIHAI} {self._service_name}"
+        self._name = self._service_name
+
+    @property
+    def device_info(self) -> DeviceInfo:
```

Comment:
Using _attr_device_info was making MyPy upset, conflict with base class Entity.  This seems to work...